airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/standard/PostgresStandardStateManager.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.postgres.standard;
+
+import com.google.common.collect.Lists;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.source.postgres.internal.models.InternalModels.StateType;
+import io.airbyte.integrations.source.postgres.internal.models.StandardStatus;
+import io.airbyte.integrations.source.relationaldb.CursorInfo;
+import io.airbyte.integrations.source.relationaldb.models.DbState;
+import io.airbyte.integrations.source.relationaldb.state.StreamStateManager;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair;
+import io.airbyte.protocol.models.v0.AirbyteStreamState;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.v0.StreamDescriptor;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PostgresStandardStateManager extends StreamStateManager {
```

Comment:
Can you add a comment on what this extends from the stream state manager? And create a ticket to add this logic to the streamstatemanager before we work on MySQL?

Reply:
I think this can be added to the `StreamStateManager` itself since it's only used by Postgres and MySQL - but we can refactor that later as well. IIUC, this is basically adding the version and state_type to the state message correct? 

Side note (and sorry if you've already done this) but we should test the upgrade and downgrade cases : 
1. Upgrade case : Old cursor state when using the new connector works as expected and gets upgraded to new state. 
2. Downgrade case : New cursor state when using an old (rolled back) connector works as expected and gets downgraded to the old state